resonance/pipe/baseline.py

```diff
@@ -0,0 +1,35 @@
+import resonance.si as si
+import resonance.db as db
+from resonance.internal import declare_transformation, Processor
+import numpy as np
+
+
+@declare_transformation
+class baseline(Processor):
+
+    def prepare(self, input, beginOffset, endOffset):
+        if not isinstance(input, db.Window):
+            raise Exception("BaseLine processor: received data block is not a window.")
+
+        if input.SI.samples < beginOffset:
+            raise Exception("BaseLine processor: offset value should not exceed the length of the window.")
+
+        if (input.SI.samples < endOffset) or (input.SI.samples < beginOffset):
+            raise Exception("BaseLine processor: the number of samples for averaging should not exceed the length of "
+                            "the window.")
+
+        self._si = si.Window(channels=input.SI.channels, samples=input.SI.samples, samplingRate=input.SI.samplingRate)
+        self._beginOffset = beginOffset
+        self._endOffset = endOffset
+
+        return self._si
+
+    def online(self, input):
+        wndBuffer = input
+
+        for channel in wndBuffer:
+            for sample in channel:
+                sample -= channel.average(range(self._beginOffset, self._endOffset))
```

Comment:
this modifies in place the window, should be a copy (add test case covering this)
Bogdan uses some method from scipy, you can reuse it if it fits here